frontend/app/player/web/MessageManager.ts

```diff
@@ -227,26 +226,21 @@ export default class MessageManager {
     this.setMessagesLoading(true)
     this.waitingForFiles = true
 
-    try {
-      if (this.session.domURL && this.session.domURL.length > 0) {
-        await loadFiles(this.session.domURL, createNewParser())
-        this.onFileReadSuccess()
-      } else {
-        const file = await requestEFSDom(this.session.sessionId)
-        const parser = createNewParser(false)
-        await parser(file)
-      }
-    } catch (e) {
-        console.error('Cant get session replay file:', e)
-        try {
-          // back compat with old mobsUrl
-          await loadFiles(this.session.mobsUrl, createNewParser(false))
-        } catch (e) {
-          this.onFileReadFailed(e)
-        }
-    } finally {
-      this.onFileReadFinally()
-    }
+    let fileReadPromise = this.session.domURL && this.session.domURL.length > 0
+      ? loadFiles(this.session.domURL, createNewParser())
+      : Promise.reject()
+    fileReadPromise
+    // EFS fallback
+    .catch(() => requestEFSDom(this.session.sessionId).then(createNewParser(false)))
+    // old url fallback
+    .catch(e => {
+      logger.error('Can not get normal session replay file:', e)
+      // back compat fallback to an old mobsUrl
+      return loadFiles(this.session.mobsUrl, createNewParser(false))
+    })
+    .then(this.onFileReadSuccess)
```

Comment:
woodoo magic tbh, why we have loadFiles(url, parser) and loadEFSFiles(url).then(parser), lets do same for both and remove one .then?

Reply:
Those are differrent. EFS is one file, that is followed by one chunk of data. The `loadFiles` downloads several files sequentially.
I do not want to put more code coupling when it is not necessary.